src/cascadia/TerminalApp/TerminalWindow.cpp

```diff
@@ -0,0 +1,1262 @@
+                }
+            }
+
+            _root->CloseWindow(false);
+        }
+    }
+
+    winrt::TerminalApp::TaskbarState TerminalWindow::TaskbarState()
+    {
+        if (_root)
+        {
+            return _root->TaskbarState();
+        }
+        return {};
+    }
+
+    winrt::Windows::UI::Xaml::Media::Brush TerminalWindow::TitlebarBrush()
+    {
+        if (_root)
+        {
+            return _root->TitlebarBrush();
+        }
+        return { nullptr };
+    }
+    void TerminalWindow::WindowActivated(const bool activated)
+    {
+        if (_root)
+        {
+            _root->WindowActivated(activated);
+        }
+    }
+
+    // Method Description:
+    // - Returns true if we should exit the application before even starting the
+    //   window. We might want to do this if we're displaying an error message or
+    //   the version string, or if we want to open the settings file.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - true iff we should exit the application before even starting the window
+    bool TerminalWindow::ShouldExitEarly()
+    {
+        return _appArgs.ShouldExitEarly();
+    }
+
+    bool TerminalWindow::FocusMode() const
+    {
+        return _root ? _root->FocusMode() : false;
+    }
+
+    bool TerminalWindow::Fullscreen() const
+    {
+        return _root ? _root->Fullscreen() : false;
+    }
+
+    void TerminalWindow::Maximized(bool newMaximized)
+    {
+        if (_root)
+        {
+            _root->Maximized(newMaximized);
+        }
+    }
+
+    bool TerminalWindow::AlwaysOnTop() const
+    {
+        return _root ? _root->AlwaysOnTop() : false;
+    }
+
+    void TerminalWindow::SetSettingsStartupArgs(const std::vector<ActionAndArgs>& actions)
+    {
+        for (const auto& action : actions)
+        {
+            _settingsStartupArgs.push_back(action);
+        }
+        _gotSettingsStartupActions = true;
+    }
+
+    bool TerminalWindow::HasCommandlineArguments() const noexcept
+    {
+        return _hasCommandLineArguments;
+    }
+
+    // Method Description:
+    // - Sets the initial commandline to process on startup, and attempts to
+    //   parse it. Commands will be parsed into a list of ShortcutActions that
+    //   will be processed on TerminalPage::Create().
+    // - This function will have no effective result after Create() is called.
+    // - This function returns 0, unless a there was a non-zero result from
+    //   trying to parse one of the commands provided. In that case, no commands
+    //   after the failing command will be parsed, and the non-zero code
+    //   returned.
+    // Arguments:
+    // - args: an array of strings to process as a commandline. These args can contain spaces
+    // Return Value:
+    // - the result of the first command who's parsing returned a non-zero code,
+    //   or 0. (see TerminalWindow::_ParseArgs)
+    int32_t TerminalWindow::SetStartupCommandline(array_view<const winrt::hstring> args)

```

Comment:
command line -> app logic?

Reply:
resolving as a part of the "commandline args are per-window" conversation